src/NBitcoin/ChainedHeader.cs

```diff
@@ -578,6 +578,29 @@ public ChainedHeader GetAncestor(int ancestorHeight)
             return walk;
         }
 
+        /// <summary>
+        /// Convert the current <see cref="ChainedHeader" /> and <paramref name="chainedHeader"/> to an array
+        /// of consecutive headers, both items are included in the array.
+        /// </summary>
+        /// <returns>Array of consecutive headers.</returns>
```

Comment:
The whole description is confusing as well as the way this method should be used. I would expect the method to accept the count parameter which would say how many headers you want to get. From the comment it is not obvious what the result is supposed to be. So I would highly recommend to change the signature of the method so that it accept the count parameter.

Reply:
Ah you mean instead of passing a chainedHeader pass a count?

Reply:
Yes

Reply:
Then I wont be able to do this
```
            if(currentHeader != chainedHeader)
                throw new NotSupportedException("Header must be on the same chain.");
```

Reply:
Yes and it won't be missed.